src/components/dateTimePicker/index.js

```diff
@@ -90,17 +92,23 @@ class DateTimePicker extends BaseComponent {
   constructor(props) {
     super(props);
 
-    const initialValue = props.value;
-    this.chosenDate = initialValue;
+    this.chosenDate = props.value;
 
     this.state = {
       showExpandableOverlay: false,
-      value: initialValue
+      prevValue: props.value,
+      value: props.value
     };
   }
 
-  generateStyles() {
-    this.styles = createStyles(this.props);
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.value !== prevState.prevValue) {
+      return {
+        prevValue: prevState.value,
+        value: nextProps.value
+      };
+    }
+    return null;
   }
```

Comment:
There's no need to keep the value in state if the component is controlled.

Reply:
I need to keep an internal state for the onChange